src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForMultipleBuildingBlocks.cs

```diff
@@ -0,0 +1,135 @@
+            IInteractionTaskContext interactionTaskContext)
+         {
+            _interactionTasksForEventGroupBuildingBlock = interactionTasksForEventGroupBuildingBlock;
+            _interactionTasksForInitialConditionBuildingBlock = interactionTasksForInitialConditionBuildingBlock;
+            _interactionTasksForMoleculeBuildingBlock = interactionTasksForMoleculeBuildingBlock;
+            _interactionTasksForPassiveTransportBuildingBlock = interactionTasksForPassiveTransportBuildingBlock;
+            _interactionTasksForParameterValues = interactionTasksForParameterValues;
+            _interactionTasksForMobiReactionBuildingBlock = interactionTasksForMobiReactionBuildingBlock;
+            _interactionTasksForMobiSpatialStructureBuildingBlock = interactionTasksForMobiSpatialStructureBuildingBlock;
+            _interactionTasksForObserverBuildingBlock = interactionTasksForObserverBuildingBlock;
+            _interactionTasksForIndividualBuildingBlock = interactionTasksForIndividualBuildingBlock;
+            _interactionTasksForExpressionProfileBuildingBlock = interactionTasksForExpressionProfileBuildingBlock;
+            _interactionTaskContext = interactionTaskContext;
+         }
+
+         public void RemoveBuildingBlocks(IReadOnlyList<IBuildingBlock> buildingBlocksToRemove)
+         {
+            if (_interactionTaskContext.DialogCreator.MessageBoxYesNo(AppConstants.Dialog.RemoveMultipleBuildingBlocks) != ViewResult.Yes)
+               return;
+
+            var referringSimulationsAndBuildingBlocks = new List<Tuple<IReadOnlyList<string>, string>>();
+
+            foreach (var buildingBlockToRemove in buildingBlocksToRemove)
+            {
+               var referringSimulations = _interactionTaskContext.Context.CurrentProject.SimulationsUsing(buildingBlockToRemove);
+
+               if (referringSimulations.Any())
+                  referringSimulationsAndBuildingBlocks.Add(new Tuple<IReadOnlyList<string>, string>(referringSimulations.AllNames(), buildingBlockToRemove.Name));
+               else
+               {
+                  removeBuildingBlock(buildingBlockToRemove);
+               }
+            }
+
+            if (referringSimulationsAndBuildingBlocks.Any())
+            {
+               var messageBuilder = AppConstants.ListOfBuildingBlocksNotRemoved(referringSimulationsAndBuildingBlocks);
+               _interactionTaskContext.DialogCreator.MessageBoxInfo(messageBuilder.ToString());
+            }
+         }
+
+         private void removeBuildingBlock(IBuildingBlock buildingBlockToRemove)
+         {
+            switch (buildingBlockToRemove)
+            {
+               case EventGroupBuildingBlock eventGroupBuildingBlock:
+                  _interactionTaskContext.Context.AddToHistory(_interactionTasksForEventGroupBuildingBlock.Remove(eventGroupBuildingBlock, eventGroupBuildingBlock.Module, eventGroupBuildingBlock, silent: true));
```

Comment:
This probably adds all the remove commands as separate entries in the history. Instead of that, can you add them to a macro command and then add the macro command in the history? You'd probably capture all the commands in the foreach above.

Reply:
The CI build has failed